# Project 7 -- Lukas Anell

In [1]:
# /anvil/projects/tdm/data/social_media_addiction/social_media.csv
# /anvil/projects/tdm/data/starwars/characters.csv

## Question 1

In [1]:
need_coffee = True
if need_coffee:
    print("Yes! Grab a coffee first...")
else:
    print("Wow, you must be a superhero! No coffee needed")

Yes! Grab a coffee first...


In [2]:
need_coffee = False

if need_coffee:
    print("Yes! Grab a coffee first...")
else:
    print("Wow, you must be a superhero! No coffee needed")

Wow, you must be a superhero! No coffee needed


In [11]:
def get_lightsaber_color():
    color = input("Enter a lightsaber color: ").strip().lower()
    if color == "green":
        print("Jedi Consular")
    elif color == "blue":
        print("Jedi Guardian")
    elif color == "purple":
        print("Unique balance")
    elif color == "red":
        print("Sith Lord")
    else:
        print("Rare Occurrence")

In [12]:
get_lightsaber_color()

Enter a lightsaber color:  green


Jedi Consular


In [13]:
get_lightsaber_color()

Enter a lightsaber color:  purple


Unique balance


In [14]:
get_lightsaber_color()

Enter a lightsaber color:  orange


Rare Occurrence


In [6]:
lightsaber_roles = {
    "green": "Jedi Consular",
    "blue": "Jedi Guardian",
    "purple": "Unique Balance",
    "red": "Sith Lord"
}

In [7]:
color = input("Enter a lightsaber color: ").strip().lower()
print('This means', lightsaber_roles.get(color, "Rare Occurrence"))

Enter a lightsaber color:  red


This means Sith Lord


In [8]:
color = input("Enter a lightsaber color: ").strip().lower()
print('This means', lightsaber_roles.get(color, "Rare Occurrence"))

Enter a lightsaber color:  green


This means Jedi Consular


In [10]:
color = input("Enter a lightsaber color: ").strip().lower()
print('This means', lightsaber_roles.get(color, "Rare Occurrence"))

Enter a lightsaber color:  brown


This means Rare Occurrence


I kept pressing shift + enter when inputting my lightsaber colors and was very confused about why it wouldn't print out my result, but I realized that pressing shift enter runs the whole cell again rather than just inputting the color I put.

1.1: See above.

1.2: See above.

1.3: See above.

## Question 2

In [15]:
import pandas as pd

In [16]:
myDF = pd.read_csv("/anvil/projects/tdm/data/social_media_addiction/social_media.csv")
myDF.head()

,Student_ID,Age,Gender,Academic_Level,Country,Avg_Daily_Usage_Hours,Most_Used_Platform,Affects_Academic_Performance,Sleep_Hours_Per_Night,Mental_Health_Score,Relationship_Status,Conflicts_Over_Social_Media,Addicted_Score
0,1,19,Female,Undergraduate,Bangladesh,5.2,Instagram,Yes,6.5,6,In Relationship,3,8
1,2,22,Male,Graduate,India,2.1,Twitter,No,7.5,8,Single,0,3
2,3,20,Female,Undergraduate,USA,6.0,TikTok,Yes,5.0,5,Complicated,4,9
3,4,18,Male,High School,UK,3.0,YouTube,No,7.0,7,Single,1,4
4,5,21,Male,Graduate,Canada,4.5,Facebook,Yes,6.0,6,In Relationship,2,7


In [23]:
max(myDF["Sleep_Hours_Per_Night"])

9.6

In [24]:
max(myDF["Avg_Daily_Usage_Hours"])

8.5

In [17]:
import numpy as np

In [18]:
myDF['Age_Status'] = np.where(myDF['Age'] > 22, "Old",
                             np.where(myDF['Age'] < 20, "Young",
                                     "Middle"))
myDF["Age_Status"].head()

0     Young
1    Middle
2    Middle
3     Young
4    Middle
Name: Age_Status, dtype: object

In [19]:
myDF["Status"] = np.where(myDF["Avg_Daily_Usage_Hours"] > myDF["Sleep_Hours_Per_Night"], "That's not very good...", np.where(myDF["Avg_Daily_Usage_Hours"] < myDF["Sleep_Hours_Per_Night"], "Pretty good!", "You're very average."))

In [20]:
myDF.head()

,Student_ID,Age,Gender,Academic_Level,Country,Avg_Daily_Usage_Hours,Most_Used_Platform,Affects_Academic_Performance,Sleep_Hours_Per_Night,Mental_Health_Score,Relationship_Status,Conflicts_Over_Social_Media,Addicted_Score,Age_Status,Status
0,1,19,Female,Undergraduate,Bangladesh,5.2,Instagram,Yes,6.5,6,In Relationship,3,8,Young,Pretty good!
1,2,22,Male,Graduate,India,2.1,Twitter,No,7.5,8,Single,0,3,Middle,Pretty good!
2,3,20,Female,Undergraduate,USA,6.0,TikTok,Yes,5.0,5,Complicated,4,9,Middle,That's not very good...
3,4,18,Male,High School,UK,3.0,YouTube,No,7.0,7,Single,1,4,Young,Pretty good!
4,5,21,Male,Graduate,Canada,4.5,Facebook,Yes,6.0,6,In Relationship,2,7,Middle,Pretty good!


In [21]:
myDF["Status"].value_counts()

Status
Pretty good!               536
That's not very good...    147
You're very average.        22
Name: count, dtype: int64

In [22]:
myDF[["Avg_Daily_Usage_Hours", "Sleep_Hours_Per_Night", "Status"]][30:40]

,Avg_Daily_Usage_Hours,Sleep_Hours_Per_Night,Status
30,3.6,6.7,Pretty good!
31,4.8,5.7,Pretty good!
32,5.4,5.4,You're very average.
33,2.6,7.2,Pretty good!
34,4.7,5.8,Pretty good!
35,5.6,5.5,That's not very good...
36,3.1,6.8,Pretty good!
37,6.2,5.1,That's not very good...
38,3.9,6.5,Pretty good!
39,4.5,5.8,Pretty good!


I enjoyed making my own descriptions for the Status column.

2.1: 9.6 hours was the longest sleep time per night, and 8.6 hours was the longest social media time.

2.2: Having Avg_Daily_Usage_Hours be less than Sleep_Hours_Per_Night was most common.

2.3: See above.

## Question 3

In [25]:
screen_time = 10
while screen_time > 2:
    print(f"Screen time: {screen_time} hours")
    screen_time -= 1
print(f"Final screen_time: {screen_time}")

Screen time: 10 hours
Screen time: 9 hours
Screen time: 8 hours
Screen time: 7 hours
Screen time: 6 hours
Screen time: 5 hours
Screen time: 4 hours
Screen time: 3 hours
Final screen_time: 2


In [27]:
sleep_time = 2
while sleep_time < 10:
    print(f"Sleep time: {sleep_time} hours")
    sleep_time += 1
print(f"Final sleep_time: {sleep_time}")

Sleep time: 2 hours
Sleep time: 3 hours
Sleep time: 4 hours
Sleep time: 5 hours
Sleep time: 6 hours
Sleep time: 7 hours
Sleep time: 8 hours
Sleep time: 9 hours
Final sleep_time: 10


In [28]:
screen_time = 10
sleep_time = 2
while screen_time > 2:
    print(f"Screen time: {screen_time} hours, Sleep time: {sleep_time} hours")
    screen_time -= 0.5
    sleep_time += 0.5
print(f"Final screen_time: {screen_time}")

Screen time: 10 hours, Sleep time: 2 hours
Screen time: 9.5 hours, Sleep time: 2.5 hours
Screen time: 9.0 hours, Sleep time: 3.0 hours
Screen time: 8.5 hours, Sleep time: 3.5 hours
Screen time: 8.0 hours, Sleep time: 4.0 hours
Screen time: 7.5 hours, Sleep time: 4.5 hours
Screen time: 7.0 hours, Sleep time: 5.0 hours
Screen time: 6.5 hours, Sleep time: 5.5 hours
Screen time: 6.0 hours, Sleep time: 6.0 hours
Screen time: 5.5 hours, Sleep time: 6.5 hours
Screen time: 5.0 hours, Sleep time: 7.0 hours
Screen time: 4.5 hours, Sleep time: 7.5 hours
Screen time: 4.0 hours, Sleep time: 8.0 hours
Screen time: 3.5 hours, Sleep time: 8.5 hours
Screen time: 3.0 hours, Sleep time: 9.0 hours
Screen time: 2.5 hours, Sleep time: 9.5 hours
Final screen_time: 2.0


3.1: See above.

3.2: See above.

## Question 4

In [29]:
characters = pd.read_csv("/anvil/projects/tdm/data/starwars/characters.csv")
characters.head()

,id,name,species,gender,height,weight,hair_color,eye_color,skin_color,year_born,homeworld,year_died,description
0,1,Luke Skywalker,Human,Male,1.72,77.0,Blond,Blue,Light,19.0,Tatooine,34.0,The main protagonist of the original trilogy.
1,2,Leia Organa,Human,Female,1.50,49.0,Brown,Brown,Light,19.0,Alderaan,35.0,A leader in the Rebel Alliance and twin sister...
2,3,Darth Vader,Human,Male,2.02,136.0,NaN,Yellow,Pale,41.0,Tatooine,4.0,The Sith Lord formerly known as Anakin Skywalker.
3,4,Yoda,Yoda's species,Male,0.66,17.0,White,Brown,Green,896.0,Unknown,4.0,A wise and powerful Jedi Master.
4,5,Han Solo,Human,Male,1.80,80.0,Brown,Hazel,Light,29.0,Corellia,34.0,A smuggler turned hero in the Rebel Alliance.


In [36]:
i = 0
char_count = 0

while char_count < 20:
    if characters.loc[i, "species"] == "Human":
        print(f"Character number {i} is a human")
    else:
        print(f"Character number {i} is NOT a human")

    i += 1
    char_count += 1

Character number 0 is a human
Character number 1 is a human
Character number 2 is a human
Character number 3 is NOT a human
Character number 4 is a human
Character number 5 is NOT a human
Character number 6 is a human
Character number 7 is a human
Character number 8 is a human
Character number 9 is a human
Character number 10 is a human
Character number 11 is a human
Character number 12 is NOT a human
Character number 13 is a human
Character number 14 is a human
Character number 15 is a human
Character number 16 is a human
Character number 17 is a human
Character number 18 is NOT a human
Character number 19 is NOT a human


In [37]:
human_count = 0
nonhuman_count = 0

i = 0
char_count = 0

while char_count < 20:
    if characters.loc[i, "species"] == "Human":
        human_count += 1
    else:
        nonhuman_count += 1

    i += 1
    char_count += 1

print(f"Humans: {human_count}")
print(f"Non-humans: {nonhuman_count}")

Humans: 15
Non-humans: 5


4.1: See above.

4.2: See above.

4.3: 15 were human, 5 were not human. If there were fewer than 20 rows, it would most likely return an index out of bounds error.

## Question 5

In [38]:
my_list = [1, 4, 5, 2, 8, 4, 6, 3, 9, 3, 2, 2, 4, 1]
i = 0
total_count = 0

while True:
    my_score = my_list[i]
    print(f"{total_count} + {my_score} = ", end="")
    total_count += my_score
    print(total_count)

    if total_count > 40:
        print("You win!!!!!!!")
        break

    i += 1

0 + 1 = 1
1 + 4 = 5
5 + 5 = 10
10 + 2 = 12
12 + 8 = 20
20 + 4 = 24
24 + 6 = 30
30 + 3 = 33
33 + 9 = 42
You win!!!!!!!


In [39]:
myDF["Mental_Health_Score"].head()

0    6
1    8
2    5
3    7
4    6
Name: Mental_Health_Score, dtype: int64

In [41]:
i = 0
total_count = 0

print(f"Starting mental health score is {total_count}\n")

while True:
    student_score = myDF["Mental_Health_Score"][i]
    print(f"Mental health of student {i} is {student_score}")
    total_count = total_count + student_score
    print(f"Current mental health score is {total_count}\n")

    if total_count >= 100:
        print("All done!!!!")
        break

    i = i + 1

Starting mental health score is 0

Mental health of student 0 is 6
Current mental health score is 6

Mental health of student 1 is 8
Current mental health score is 14

Mental health of student 2 is 5
Current mental health score is 19

Mental health of student 3 is 7
Current mental health score is 26

Mental health of student 4 is 6
Current mental health score is 32

Mental health of student 5 is 4
Current mental health score is 36

Mental health of student 6 is 9
Current mental health score is 45

Mental health of student 7 is 6
Current mental health score is 51

Mental health of student 8 is 7
Current mental health score is 58

Mental health of student 9 is 7
Current mental health score is 65

Mental health of student 10 is 5
Current mental health score is 70

Mental health of student 11 is 6
Current mental health score is 76

Mental health of student 12 is 8
Current mental health score is 84

Mental health of student 13 is 5
Current mental health score is 89

Mental health of student



5.1: `while` loops until its boolean condition is false. This doesn't happen with `while True` because True is never equal to false, so you have to halt the loop yourself.

5.2: See above.

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.

https://www.purdue.edu/odos/osrr/honor-pledge/
